### PR TITLE
util: add test for old version cgroup

### LINKED ISSUE #41808: disable the scheduler when run under too old cgroups
## Enhancement

If you run the TiDB in the container with a kernel version lower than 4.7, you can not get the CPU usage of this container from groups. We get the CPU usage from the usage_user and usage_sys of cgroups. But those values are created on Linux 4.7. 

If we find the missing path of the cgroups path, we will stop the scheduler of the resource manager so that it is still able to work but it can not tune the concurrency.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #41808

Problem Summary:

Now We will get zero CPU usage if we meet the old cgroup. Because we will not start the CPU observer to watch the CPU usage.

### What is changed and how it works?

add the test and enable the goleak for it. if a bug is here, it will happen goleak.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

**Comment by hawkingrei** [comment]
@Benjamin2037 @zimulala PTAL

File: util/cpu/cpu_test.go [hunk 1]
```diff
@@ -44,13 +48,49 @@ func TestCPUValue(t *testing.T) {
...
+	}
+	observer.Start()
+	for n := 0; n < 10; n++ {
+		time.Sleep(1 * time.Second)
+		value, unsupported := cpu.GetCPUUsage()
+		require.True(t, unsupported)
+		require.Equal(t, value, 0.0)
+	}
+	sch := scheduler.CPUScheduler{}
+	require.Equal(t, scheduler.Hold, sch.Tune(util.UNKNOWN, util.NewMockGPool("test")))
```
Comment (hawkingrei) [inline util/cpu/cpu_test.go:91]: The test of CPUScheduler is here.

File: resourcemanager/util/mock_gpool.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ func (*MockGPool) Tune(_ int) {
 
 // LastTunerTs is only for test
 func (*MockGPool) LastTunerTs() time.Time {
-	panic("implement me")
+	return time.Now().Add(-1 * 10 * time.Second)
```
Comment (Benjamin2037) [inline resourcemanager/util/mock_gpool.go:41]: Why here is -1？

Comment (hawkingrei) [reply resourcemanager/util/mock_gpool.go:41]: The last Tuner timestamp must be before now.

Comment (hawkingrei) [reply resourcemanager/util/mock_gpool.go:41]: BTW, the api of time is in golang .


[func (t Time) Add(d Duration) Time](https://pkg.go.dev/time#Time.Add)
func (t [Time](https://pkg.go.dev/time#Time)) Sub(u [Time](https://pkg.go.dev/time#Time)) [Duration](https://pkg.go.dev/time#Duration)

**Comment by hawkingrei** [comment]
@Benjamin2037 PTAL

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**